File: clients/csharp/examples/tables-create-type-exists.cs
using DataStax.AstraDB.DataApi;
using DataStax.AstraDB.DataApi.Core;
using DataStax.AstraDB.DataApi.Tables;

namespace Examples;

// Define the user-defined type
[UserDefinedType("member")]
public class Member
{
  [ColumnName("name")]
  public string? Name { get; set; }

  [ColumnName("is_active")]
  public bool? IsActive { get; set; }

  [ColumnName("date_joined")]
  public DateOnly? DateJoined { get; set; }
};

public class Program
{
  static async Task Main()
  {
    // Get an existing table
    var client = new DataAPIClient();
    var database = client.GetDatabase(
      "**API_ENDPOINT**",
      "**APPLICATION_TOKEN**"
    );

    // Create a user-defined type
    await database.CreateTypeAsync<Member>(
      new CreateTypeOptions() { IfNotExists = true }
    );
  }
}

File: clients/java/examples/tables-create-type-exists.java
import com.datastax.astra.client.DataAPIClient;
import com.datastax.astra.client.databases.Database;
import com.datastax.astra.client.tables.commands.options.CreateTypeOptions;
import com.datastax.astra.client.tables.definition.types.TableUserDefinedTypeDefinition;

public class Example {

  public static void main(String[] args) {
    // Get a database
    Database database = new DataAPIClient("**APPLICATION_TOKEN**").getDatabase("**API_ENDPOINT**");

    // Create a user-defined type
    TableUserDefinedTypeDefinition typeDefinition =
        new TableUserDefinedTypeDefinition()
            .addFieldText("name")
            .addFieldBoolean("is_active")
            .addFieldDate("date_joined");
    database.createType("member", typeDefinition, new CreateTypeOptions().ifNotExists(true));
  }
}
